Cabal/src/Distribution/Simple/Setup.hs

```diff
@@ -1545,6 +1570,245 @@ instance Monoid HaddockFlags where
 instance Semigroup HaddockFlags where
   (<>) = gmappend
 
+-- ------------------------------------------------------------
+-- * HaddocksFlags flags
+-- ------------------------------------------------------------
+
+-- | Governs whether modules from a given interface should be visible or
+-- hidden in the Haddock generated content page.  We don't expose this
+-- functionality to the user, but simply use 'Visible' for only local packages.
+-- Visibility of modules is available since @haddock-2.26.1@.
+--
+data Visibility = Visible | Hidden
+  deriving (Eq, Show)
+
+data HaddockProjectFlags = HaddockProjectFlags {
```

Comment:
isn't this specific to cabal-install?

Reply:
`haddock` has two modes of operation: build documentation or build the main page with indexes: I think it's good to expose both through `Cabal`, but if you think that's better to move it to `cabal-install` then that's ok too (also parts of `Distribution.Simple.Haddock` will need to move there too, e.g. `createHaddockIndex` and `fromHaddockProjectFlags`).

Reply:
Will this be useful if `stack` would like to follow with their implementation?

Reply:
Keeping it in Cabal sounds good